geowebcache/core/src/main/java/org/geowebcache/storage/BlobStore.java

```diff
@@ -136,6 +207,55 @@ public boolean deleteByGridsetId(final String layerName, final String gridSetId)
 	 */
 	public boolean layerExists(String layerName);
 
+    Map<String,Optional<Map<String, String>>> getParametersMapping(String layerName);
+    
+    /**
+     * If the given layer is cached, remove 
+     * @param layer
+     * @throws StorageException
+     */
+    public default boolean purgeOrphans(TileLayer layer) throws StorageException {
+        // TODO maybe do purging based on gridset and format
+        try{
+            final List<ParameterFilter> parameterFilters = layer.getParameterFilters();
+            
+            return getParametersMapping(layer.getName()).entrySet().stream()
```

Comment:
This seems like too many nested streams and lambda, I have no idea of what it does. Could it be split in more understandable building blocks?